### DISCUSSION TITLE
What is the best way to allow an unauthenticated user to access an existing page?

### DISCUSSION THREAD

**Discussion by nextechinc** [filamentphp/filament#12264] [Help]
### Package

Panel builder

### Package Version

v3.2.52

### How can we help you?

I have an infolist that displays an invoice.  I like the way it looks when logged into filament, and would ideally like whatever solution I implement to look as close to this view as possible.  I want to be able to send a customer an e-mail with a link to their invoice, which allows them to pay it, without them having to create an account.  I've tried following the steps in the documentation to create a livewire component, and insert an infolist into that view.  This does present a non-stylized view of the invoice, but I'm unable to get the related items (invoice line items) to display in this view.  Any ideas on how best to get this infolist viewable for unauthenticated users would be welcomed.

**Comment by leandrocfe** [comment]
- create a LW component
`php artisan make:livewire Invoice`

- add the route in web.php
```php
Route::get('invoice', \App\Livewire\Invoice::class)->name('invoice');
```
- extend SimplePage and implement Infolist
```php
class Invoice extends SimplePage implements HasForms, HasInfolists
{
    use InteractsWithInfolists;
    use InteractsWithForms;

    protected static string $view = 'livewire.invoice';

    protected ?string $maxWidth = '5xl';

    public function invoiceInfolist(Infolist $infolist): Infolist
    {
        return $infolist
            ->record(...)
            ->schema([
                TextEntry::make('invoice_number'),
                ...
            ]);
    }
}
```
- render in the view
```html
<div>
    {{ $this->invoiceInfolist }}
</div>
```

**Reply by tahinuralam** [reply]
#How to get query ?

```php
->record (User::all())
```

**Comment by nextechinc** [comment]
Thanks for the response.  This is more or less where I've been stuck.  I have an Infolist in the panel that looks exactly how I want it to look in the front end.  However, with this approach to displaying it on the front end, I'm missing the relations, header actions, and styling.  I'm sure it can all be replicated, but it seems like I'm missing something.  It seems like I should be able to take what's already there and mirror it on the front end without having to completely regenerate everything.

**Comment by AllTheUserNamesAreGone** [comment]
This video might be of help.

[Laravel Daily on YouTube looking at their new invoice package](https://www.youtube.com/watch?v=h9iuXm23Xe0)

**Reply by nextechinc** [reply]
Thanks for the reply.  I leveraged the invoice package for generating PDF invoices, which I have working for send to e-mail, and as a button on the Infolist in question.  My expectation was that I could just allow that Infolist view page to be accessible on the public side, but it's looking more and more like I'll just have to rebuild it in livewire.

**Comment by nextechinc** [comment]
I spent some time with Zep, and he was able to help me come up with the solution I was looking for.  Basically, we ended up creating a separate Filament panel for customers.  Then we copied over resource in question from my original admin panel.  For the page I was looking to give unauthenticated access to, adding this to the class allows it to be viewed without authentication.

`protected static string | array $withoutRouteMiddleware = [
        Authenticate::class
    ];`
